articles/firewall-manager/migrate-to-policy.md

```diff
@@ -41,140 +42,181 @@ $NatRuleGroupPriority = 100
 Function GetApplicationProtocolsString
 {
-	Param([Object[]] $Protocols)
-	$output = ""
-	ForEach ($protocol in $Protocols) {
-		$output += $protocol.ProtocolType + ":" + $protocol.Port + ","
-	}
-	return $output.Substring(0, $output.Length - 1)
+  Param([Object[]] $Protocols)
+  $output = ""
+  ForEach ($protocol in $Protocols)
+  {
+    $output += $protocol.ProtocolType + ":" + $protocol.Port + ","
+  }
+  return $output.Substring(0, $output.Length - 1)
 }
 
 Function GetApplicationRuleCmd
 {
-	Param([Object] $ApplicationRule)
-	
-	$cmd = "New-AzFirewallPolicyApplicationRule"
-	$cmd = $cmd + " -Name " + $ApplicationRule.Name
-	$cmd = $cmd + " -SourceAddress " + $ApplicationRule.SourceAddresses
-	
-	if ($ApplicationRule.Description) {
-		$cmd = $cmd + " -Description " + $ApplicationRule.Description
-	}
-	if ($ApplicationRule.TargetFqdns) {
-		$protocols = GetApplicationProtocolsString($ApplicationRule.Protocols)
-		$cmd = $cmd + " -Protocol " + $protocols
-		$cmd = $cmd + " -TargetFqdn  " + $ApplicationRule.TargetFqdns
-	}
-	if ($ApplicationRule.FqdnTags) {
-		$cmd = $cmd + " -FqdnTag  " + $ApplicationRule.FqdnTags
-	}
-	
-	return $cmd
+  Param([Object] $ApplicationRule)
+
+  $cmd = "New-AzFirewallPolicyApplicationRule"
+  $cmd = $cmd + " -Name " + "'" + $($ApplicationRule.Name) + "'"
+
+  if ($ApplicationRule.SourceAddresses)
+  {
+    $ApplicationRule.SourceAddresses = $ApplicationRule.SourceAddresses -join ","
+    $cmd = $cmd + " -SourceAddress " + $ApplicationRule.SourceAddresses
+  }
+  elseif ($ApplicationRule.SourceIpGroups)
+  {
+    $ApplicationRule.SourceIpGroups = $ApplicationRule.SourceIpGroups -join ","
+    $cmd = $cmd + " -SourceIpGroup " + $ApplicationRule.SourceIpGroups
+  }
+
+  if ($ApplicationRule.Description)
+ {
+    $cmd = $cmd + " -Description " + "'" + $ApplicationRule.Description + "'"
+  }
+  if ($ApplicationRule.TargetFqdns)
+  {
+    $protocols = GetApplicationProtocolsString($ApplicationRule.Protocols)
+    $cmd = $cmd + " -Protocol " + $protocols
+
+    $AppRule = $($ApplicationRule.TargetFqdns) -join ","
+    $cmd = $cmd + " -TargetFqdn " + $AppRule
+
+  }
+  if ($ApplicationRule.FqdnTags)
+  {
+    $cmd = $cmd + " -FqdnTag " + "'" + $ApplicationRule.FqdnTags + "'"
+  }
+
+  return $cmd
 }
 
-If(!(Get-AzResourceGroup -Name $FirewallResourceGroup))
+If (!(Get-AzResourceGroup -Name $FirewallPolicyResourceGroup))
 {
-    New-AzResourceGroup -Name $FirewallResourceGroup -Location $FirewallPolicyLocation
+  New-AzResourceGroup -Name $FirewallPolicyResourceGroup -Location $FirewallPolicyLocation
 }
 
 $azfw = Get-AzFirewall -Name $FirewallName -ResourceGroupName $FirewallResourceGroup
+
 Write-Host "creating empty firewall policy"
-$fwp = New-AzFirewallPolicy -Name $FirewallPolicyName -ResourceGroupName $FirewallResourceGroup -Location $FirewallPolicyLocation -ThreatIntelMode $azfw.ThreatIntelMode
+$fwDnsSetting = New-AzFirewallPolicyDnsSetting -EnableProxy
```

Comment:
instead of enabling DNS proxy in every policy, we rather enabling only if $azfw.DNSEnableProxy is true